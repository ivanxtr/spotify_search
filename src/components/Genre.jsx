import React from 'react'

function Genre({
  data,
  getPlaylistByGenre,
  showGenre
}) {
  const genreView = () => {
    return data.map((item, index) => (
        <div
        key={index}
          className="genre"
          onClick={() => getPlaylistByGenre(item.id)}
        >
          <p>{ item.name }</p>
          <img src={item.icons[0].url} alt={ item.name } />
        </div>
      )
    )
  }

  return (
    <>
     <section>
      <div className={showGenre ? "genre-container" : "genre-container hide"}>
        { data ? genreView() : false }
      </div>
     </section>
    </>
  )
}

export default Genre
