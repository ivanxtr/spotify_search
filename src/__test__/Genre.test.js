import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Genre from '../components/Genre';

Enzyme.configure({ adapter: new Adapter() });

const dataFixture = [
      {
        "href" : "https://api.spotify.com/v1/browse/categories/toplists",
        "icons" : [ {
          "height" : 275,
          "url" : "https://t.scdn.co/media/derived/toplists_11160599e6a04ac5d6f2757f5511778f_0_0_275_275.jpg",
          "width" : 275
        } ],
        "id" : "toplists",
        "name" : "Top Lists"
      }, {
        "href" : "https://api.spotify.com/v1/browse/categories/at_home",
        "icons" : [ {
          "height" : null,
          "url" : "https://t.scdn.co/images/04da469dd7be4dab96659aa1fa9f0ac9.jpeg",
          "width" : null
        } ],
        "id" : "at_home",
        "name" : "At Home"
      }]

describe('Genre Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Genre />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot render', () => {
    const component = renderer.create(<Genre />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should display proper content', () => {
    const wrapper = mount(
      <Genre
        showGenre={true}
        data={dataFixture}
        getPlaylistByGenre={() => {}} 
      />);
    expect(wrapper.find('section')).toHaveLength(1)
    expect(wrapper.find('.genre-container')).toHaveLength(1)
    expect(wrapper.find('.hide')).toHaveLength(0)
    expect(wrapper.find('.genre')).toHaveLength(2)
    expect(wrapper.find('.genre p').at(0).text()).toBe('Top Lists')
    expect(wrapper.find('.genre p').at(1).text()).toBe('At Home')
  })

  it('should be hidden when empty data is recieved', () => {
    const wrapper = mount(
      <Genre
        showPlaylist={true}
        data={[]}
        filterData={() => {}} 
      />);
      expect(wrapper.find('section')).toHaveLength(1)
      expect(wrapper.find('.genre-container')).toHaveLength(1)
      expect(wrapper.find('.hide')).toHaveLength(1)
      expect(wrapper.find('.genre')).toHaveLength(0)
  })
})
