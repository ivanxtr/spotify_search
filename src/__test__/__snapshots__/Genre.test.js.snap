// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre Component snapshot render 1`] = `
<section>
  <div
    className="genre-container hide"
  />
</section>
`;
